### PR TITLE
log backup: move region-operation to a separated file

### PR DESCRIPTION
**Pull request by YuJuncen**
### What is changed and how it works?

Issue Number: ref #12715 

What's Changed:

This PR moved the handler of `ObserveOp` to a separated file, and make a separated queue for them: so operations over it won't block the main queue, and they can keep some internal status.

```commit-message
- move the handler of subscription change operation to another file.
- drop temporary files when stopped.
```

### Check List 

Tests 

- Integration test
- Manual test (add detailed scripts or steps below)

### Release note 

```release-note
none
```

### CODE REVIEW

File: components/backup-stream/src/subscription_manager.rs [hunk 1]
```diff
@@ -0,0 +1,445 @@
...
+            annotate!(err, "BUG: scan_pool closed").report(format!(
+                "during initial scanning for region {}",
+                region.get_id()
+            ));
+        }
+    }
+
+    fn find_task_by_region(&self, r: &Region) -> Option<String> {
+        self.range_router
+            .find_task_by_range(&r.start_key, &r.end_key)
```
Comment (3pointer) [inline components/backup-stream/src/subscription_manager.rs:443]: will a region include multiple task in the future?

File: components/backup-stream/src/subscription_manager.rs [hunk 2]
```diff
@@ -0,0 +1,445 @@
...
+
+    async fn refresh_resolver(&self, region: &Region) {
+        let need_refresh_all = !self.subs.try_update_region(region);
+
+        if need_refresh_all {
+            let canceled = self.subs.deregister_region(region, |_, _| true);
+            let handle = ObserveHandle::new();
+            if canceled {
+                let for_task = self.find_task_by_region(region).unwrap_or_else(|| {
+                    panic!(
```
Comment (3pointer) [inline components/backup-stream/src/subscription_manager.rs:282]: can we ignore the error and do not panic?

maybe at this time the task is dropped🤔

File: components/backup-stream/src/subscription_manager.rs (hunk 1 above)
Comment (YuJuncen) [reply components/backup-stream/src/subscription_manager.rs:443]: At that time, the rule of this should be changed.

Comment (3pointer) [reply components/backup-stream/src/subscription_manager.rs:443]: please Leave a NOTE or TODO here. to tell others we do not support multiple tasks right now, one day we implement multiple tasks we should changed it.

File: components/backup-stream/src/event_loader.rs [hunk 3]
```diff
@@ -69,8 +69,12 @@ impl PendingMemoryQuota {
 /// EventLoader transforms data from the snapshot into ApplyEvent.
 pub struct EventLoader<S: Snapshot> {
     scanner: DeltaScanner<S>,
+    // pooling the memory.
+    entry_batch: EntryBatch,
 }
 
+const ENTRY_BATCH_SIZE: usize = 1024;
```
Comment (3pointer) [inline components/backup-stream/src/event_loader.rs:76]: And also give some comments about this magic number.

File: components/backup-stream/src/subscription_manager.rs [hunk 4]
```diff
@@ -0,0 +1,445 @@
...
+            // This won't make background thread deadlock,
+            // because `shutdown` won't join the handle if the target of which is current thread.
+            drop(pool_handle);
+        })
+    }
+    tx
+}
+
+/// The default channel size.
+const MESSAGE_BUFFER_SIZE: usize = 4096;
```
Comment (3pointer) [inline components/backup-stream/src/subscription_manager.rs:117]: ditto

**Review by 3pointer — APPROVED** [review]
LGTM

**Comment by joccau** [comment]
ref https://github.com/tikv/tikv/issues/12512

**Comment by YuJuncen** [comment]
close in favor of https://github.com/tikv/tikv/pull/12976

**Pull request closed without merging**